### PR TITLE
Prevent live migration of instances with custom volumes

### PR DESCRIPTION
**Pull request by MusicDin**
Error out if instance with attached custom volumes is being live migrated.

### CODE REVIEW

File: lxd/instance_post.go [hunk 1]
```diff
@@ -618,7 +619,7 @@ func instancePostMigration(s *state.State, inst instance.Instance, newName strin
 	return nil
 }
 
-// Move a non-ceph container to another cluster node.
+// Move an instnace to another cluster node. Source and target members must be online.
```
Comment (tomponline) [inline lxd/instance_post.go:622]: Typo of `instance` and the non-ceph part in the previous comment was important please retain.

File: lxd/instance_post.go [hunk 2]
```diff
@@ -956,7 +957,25 @@ func migrateInstance(s *state.State, r *http.Request, inst instance.Instance, ta
 		return err
 	}
 
-	// Check if we are migrating a ceph-based instance.
+	// In case of live migration, ensure all instance disks can be migrated.
+	if req.Live && inst.IsRunning() {
+		volatileGet := func() map[string]string { return map[string]string{} }
+		volatileSet := func(_ map[string]string) error { return nil }
+		for deviceName, rawConfig := range inst.ExpandedDevices() {
+			dev, err := device.New(inst, s, deviceName, rawConfig, volatileGet, volatileSet)
```
Comment (tomponline) [inline lxd/instance_post.go:965]: Is there any need for device.New here, can we just use rawConfig and pass to IsRootDisk directly?

**Review by tomponline — CHANGES_REQUESTED** [review]
Please can you split your commit into logical changes, there are 3 comment changes and 1 logic change, so expect 4 commits in total. This aids with backporting. ta

File: lxd/instance_post.go (hunk 1 above)
Comment (MusicDin) [reply lxd/instance_post.go]: Yep, thanks.

File: lxd/instance_post.go (hunk 2 above)
Comment (MusicDin) [reply lxd/instance_post.go]: No. I've fixed it, thanks.

**Comment by tomponline** [comment]
Thanks @MusicDin LGTM

**Review by tomponline — APPROVED** [review]

**Pull request merged**